projects/imagenet/local_run.py

```diff
@@ -0,0 +1,188 @@
+#  Numenta Platform for Intelligent Computing (NuPIC)
+#  Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+#  with Numenta, Inc., for a separate license for this software code, the
+#  following terms and conditions apply:
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU Affero Public License version 3 as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Affero Public License for more details.
+#
+#  You should have received a copy of the GNU Affero Public License
+#  along with this program.  If not, see http://www.gnu.org/licenses.
+#
+#  http://numenta.org/licenses/
+#
+"""
+Use this script to run Imagenet Experiments on a single node using Distributed
+Data Parallel (DDP) without ray
+"""
+import argparse
+import copy
+import os
+import tempfile
+from datetime import datetime
+from functools import partial
+
+import torch
+import torch.multiprocessing as multiprocessing
+
+from nupic.research.frameworks.pytorch.imagenet import (
+    ImagenetExperiment,
+    imagenet_run,
+    mixins,
+)
+from nupic.research.frameworks.sigopt.sigopt_experiment import SigOptImagenetExperiment
+
+multiprocessing.set_start_method("spawn", force=True)
+
+
+def insert_experiment_mixin(config, mixin):
+    experiment_class = config["experiment_class"]
+
+    class Cls(mixin, experiment_class):
+        pass
+
+    Cls.__name__ = f"{mixin.__name__}{experiment_class.__name__}"
+    config["experiment_class"] = Cls
+
+
+def create_trials(config):
+    """
+    Create trial configuration for each trial variant evaluating 'ray.tune'
+    functions (grid_search, sample_from, ...) into its final values and
+    creating the local and log dir for each trial
+
+    :param config: Ray tune configuration with 'ray.tune' functions
+    :return: list of dict for each trial configuration variant
+    """
+    from nupic.research.support.ray_utils import generate_trial_variants
```

Comment:
Do we need all this? Seems like overkill and replicating ray tune. All we do is use tune to get a random seed. Why don't we just move away from using ray tune altogether in our regression tests? We could just use regular random() to figure out the seed.

Reply:
We will not use this script for regression tests. The regression tests will use the `imagenet_run.run(config,logger=None, on_checkpoint=None)` API from the unit test using a custom logger to validate the results without any ray or ray.tune dependencies.

This script is just a direct replacement to `run.py` to serve as example.